core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -283,8 +283,39 @@ class BrokerTopicMetrics(name: Option[String], configOpt: java.util.Optional[Kaf
       meter()
   }
 
+  case class GaugeWrapper(metricType: String, brokerTopicAggregatedMetric: BrokerTopicAggregatedMetric) {
+    @volatile private var lazyGauge: Gauge[Long] = _
+    private val gaugeLock = new Object
+
+    def gauge(): Gauge[Long] = {
```

Comment:
When do we initialize the gauge for topic with tags?

Reply:
Good question, I believe this is a leftover from the meters. I have removed it. I have further added to an integration test. To be honest, however, I haven't found a nice way to simulate an upload to remote storage being blocked to showcase the increase in this value via a test.

Reply:
Nope, it is no longer needed, I have since removed it 😊